File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    int len = elements.size();
    
    vector<int> elements_2(elements.begin(), elements.end());
    elements_2.insert(elements_2.end(), elements.begin(), elements.end());
    unordered_set<int> element_set;
    
    
    for(int i = 1 ; i <= len ; i++){
        for(int j = 0 ; j < len ; j++){
            int sum = 0;
            for(int k = 0 ; k < i ; k++)
                sum += elements_2[j + k];
            element_set.insert(sum);
        }    
        
    }
    
    answer = element_set.size();
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] elements) {
        List<Integer>list = new ArrayList<>();
        
        int[] elementsX2 = IntStream.concat(Arrays.stream(elements),
                                           Arrays.stream(elements)).toArray();
        
        for(int i = 0 ; i < elements.length ; i++){
            int sum = 0;
            for(int j = i ; j < i + elements.length ; j++){
                sum += elementsX2[j];
                list.add(sum);
            }
        }
        
        return (int)list.stream().mapToInt(Integer::valueOf)
            .distinct().count();
    }
}